packages/x-date-pickers/src/DigitalClock/DigitalClock.tsx

```diff
@@ -173,7 +173,7 @@ export const DigitalClock = React.forwardRef(function DigitalClock<TDate extends
   });
 
   const handleValueChange = useEventCallback((newValue: TDate | null) =>
-    handleRawValueChange(newValue, 'finish'),
+    handleRawValueChange(newValue, 'finish', 'hours'),
```

Comment:
Fix to close the `DateTimePicker` immediately when the time is selected.
Basically the same flow, that `MultiSectionDigitalClock` has when clicking meridiem will select everything and close the picker.